extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -148,7 +148,20 @@ private Uni<SecurityIdentity> createSecurityIdentityWithOidcServer(RoutingContex
                 tokenUni = verifySelfSignedTokenUni(resolvedContext, request.getToken().getToken());
             }
         } else {
-            tokenUni = verifyTokenUni(resolvedContext, request.getToken().getToken());
+            // verify opaque access token with UserInfo if enabled
+            if (resolvedContext.oidcConfig != null && resolvedContext.oidcConfig.verifyOpaqueAccessTokenWithUserInfo
+                    && request.getToken() instanceof AccessTokenCredential
+                    && ((AccessTokenCredential) request.getToken()).isOpaque()) {
+                if (userInfo == null) {
+                    tokenUni = Uni.createFrom().failure(
+                            new AuthenticationFailedException("Opaque access token verification failed as user info is null."));
+                } else {
+                    // valid token verification result with empty JWT content
+                    tokenUni = Uni.createFrom().item(new TokenVerificationResult(new JsonObject(), null));
```

Comment:
Returning empty JSON object can cause the confusion if someone attempts to inject `JsonWebToken`  it should be null I guess and somewhere above, when checking the token verification result, the error should be returned if this null object is accessed when it should not, example if the source of roles is the access token - but we only have a non-null UserInfo, please set it to `null` and then we can tune null checks

Reply:
I believe it is fixed.